src/frontend/src/catalog/source_catalog.rs

```diff
@@ -93,7 +95,8 @@ impl From<&PbSource> for SourceCatalog {
             .into_iter()
             .map(Into::into)
             .collect();
-        let with_options = WithOptions::new(prost.properties.clone());
+        let properties = WithOptions::new(prost.properties.clone()).into_inner();
+        let options = WithOptions::new(prost.options.clone()).into_inner();
```

Comment:
We might need to rename the type alias now... 🥵

Reply:
Besides, `properties` and `options` are not meaningful names. Can we change them to sth like `with_options` and `encode_format_options`/`row_options`?

Reply:
How about `with_properties` and `row_options`?

Reply:
the most of fields in `option` are related to parser behavior, I prefer `encode_option` or `parser_option`

Reply:
I have been calling them `format options` for sink 😂

Reply:
Let's start a vote for `options`. Please use 👍 to pick the names you think is acceptable.

Reply:
row_options

Reply:
format_options

Reply:
encode_format_options

Reply:
parser_options

Reply:
Wait a minute, the syntax seems to be `FORMAT ... ENCODE ... (...)`, so at least it should be `format_encode_options` 🤣

Reply:
From the syntax, the options indeed look like it's for `ENCODE` 🤔.

From the parser's definition it seems to be called `row_options`, so I guess this is what it should be like when it was designed (and not only for `ENCODE`).  So might worth asking @st1page what's the best name for it.

https://github.com/risingwavelabs/risingwave/blob/b5c3f5dfd091d14184f06c5d93108604fcb1f333/src/sqlparser/src/ast/statement.rs#L338-L342



> the most of fields in option are related to parser behavior, I prefer `encode_option` or `parser_option`

I'm wondering whether `parser_options` will be confusing to users. We need to consider this in the error messages (maybe also syntax reference, but currently it also doesn't have a name)

Reply:
I think it is the historical reason...  `encode_option` LGTM. Btw, we might need some option in the `KEY ENCODE` in the future. c.c. @tabVersion 
```SQL
format upsert 
encode avro (
  schema.registry = 'http://message_queue:8081/',
  schema.registry.name.strategy = 'typo')
key encode avro (
  schema.registry = 'http://message_queue:8081/',
  schema.registry.name.strategy = 'typo')
)
```

Reply:
> I think it is the historical reason... format_encode_options LGTM.
> Btw, we might need some options in the KEY ENCODE in the future. 
+1 for `format_encode_options`
for KEY ENCODE, I think it is ok to add some fields in the options because the clause is optional in SQL and we don't expect it as strong as the value part.